src/mon/MonClient.cc

```diff
@@ -687,21 +687,22 @@ MonConnection& MonClient::_add_conn(unsigned rank, uint64_t global_id)
 
 void MonClient::_add_conns(uint64_t global_id)
 {
-  uint16_t min_priority = std::numeric_limits<uint16_t>::max();
+  map<uint16_t, vector<unsigned>> rank_by_priority;
   for (const auto& m : monmap.mon_info) {
-    if (m.second.priority < min_priority) {
-      min_priority = m.second.priority;
-    }
+    rank_by_priority[m.second.priority].push_back(monmap.get_rank(m.first));
   }
   vector<unsigned> ranks;
```

Comment:
Maybe what we really want is to make our selection of a rank proportional to it's priority.. i.e., a weight.  We would add ``weight`` instances of the rank to the vector, shuffle, and choose the first (unique) N of them.  Or maybe std:: has something to do weighted shuffle/selection).

Reply:
Yup, that was __exact__ my initial thought.

Reply:
first, i don't understand why we want to rewrite the existing shuffling logic in a different way. and the weight and priority are different things in SRV record. i'd suggest avoid taking one as another.